phpstan.neon.dist

```diff
@@ -1,2 +1,4 @@
 includes:
 	- phpstan-baseline.neon
+	- phpstan-docs-baseline.neon
+	- phpstan-unsupported-baseline.neon
```

Comment:
- the docs baseline is just issues in `docs/` so nothing important but fixing them requires diving into other methods, which I will do anyway in future PRs.

- the unsupported baseline refers to the PHPStan unsupported feature I mentioned [here](https://github.com/loophp/collection/pull/96/files#r638333789). It's limited to the `unpack` operation and I expect we'll be able to get rid of it once `@template T of array` is supported

Reply:
Nice and clean!

This is why I was using `@psalm-template`

By the way, what do you think about removing that `of array-key` ?
As this is a lazy library, the TKey can be of any type, so, I don't think I'm wrong by saying we can get rid of it. WDYT ?

Reply:
I think you're right we could get rid of `array-key` but I don't think it can only be done in one place because it will probably result in static analysis errors.

I'll look to prepare a PR after this to remove it everywhere and see how many errors we get 😄

Reply:
Yeah I did it in #95 maybe it could be a starting point.